### ISSUE TITLE
LiKI module build fails on Fedora 28

### ISSUE THREAD

**Issue by cvoltz** [HewlettPackard/LinuxKI#9]
When installing the RPM on Fedora 28, the LiKI module build fails:
``` bash
$ sudo rpm -ivh rpms/linuxki-5.4-1.noarch.rpm 
Preparing...                          ################################# [100%]
Updating / installing...
   1:linuxki-5.4-1                    ################################# [100%]
        Finishing installation of the Linux KI Toolset ...
        Unable to find compatible module - attempting to build from source ...
hostkernel=4.17.2-200.fc28.x86_64
hostdistro=fedora
hostversion=28
EXTRA_CFLAGS=-g -O3 -Wall -DHARDCLOCK_STACK_SKIP=0 -DOTHER_STACK_SKIP=2
make -C /lib/modules/4.17.2-200.fc28.x86_64/build  M=/tmp/tmp.quafAlVvP1   modules  
make[1]: Entering directory '/usr/src/kernels/4.17.2-200.fc28.x86_64'
hostkernel=4.17.2-200.fc28.x86_64
hostdistro=fedora
hostversion=28
EXTRA_CFLAGS=-g -O3 -Wall -DHARDCLOCK_STACK_SKIP=0 -DOTHER_STACK_SKIP=2
Building with KERNELRELEASE = 4.17.2-200.fc28.x86_64
  CC [M]  /tmp/tmp.quafAlVvP1/likit.o
/tmp/tmp.quafAlVvP1/likit.c: In function ‘syscall_exit_trace’:
/tmp/tmp.quafAlVvP1/likit.c:4116:59: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
                                                           ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4116:59: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4116:7: error: too many arguments to function ‘sock->ops->getname’
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
       ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4128:61: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
    if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
                                                             ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4128:61: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4128:8: error: too many arguments to function ‘sock->ops->getname’
    if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
        ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4223:61: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
    if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
                                                             ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4223:61: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4223:8: error: too many arguments to function ‘sock->ops->getname’
    if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
        ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4228:60: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
    if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
                                                            ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4228:60: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4228:8: error: too many arguments to function ‘sock->ops->getname’
    if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
        ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4349:60: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
   if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
                                                            ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4349:60: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4349:7: error: too many arguments to function ‘sock->ops->getname’
   if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
       ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4354:59: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
                                                           ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4354:59: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4354:7: error: too many arguments to function ‘sock->ops->getname’
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
       ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4419:60: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
   if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
                                                            ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4419:60: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4419:7: error: too many arguments to function ‘sock->ops->getname’
   if (sock->ops->getname(sock, (struct sockaddr *)&remote, &remlen, 1)) {
       ^~~~
/tmp/tmp.quafAlVvP1/likit.c:4424:59: warning: passing argument 3 of ‘sock->ops->getname’ makes integer from pointer without a cast [-Wint-conversion]
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
                                                           ^~~~~~~
/tmp/tmp.quafAlVvP1/likit.c:4424:59: note: expected ‘int’ but argument is of type ‘int *’
/tmp/tmp.quafAlVvP1/likit.c:4424:7: error: too many arguments to function ‘sock->ops->getname’
   if (sock->ops->getname(sock, (struct sockaddr *)&local, &loclen, 0)) {
       ^~~~
make[2]: *** [scripts/Makefile.build:319: /tmp/tmp.quafAlVvP1/likit.o] Error 1
make[1]: *** [Makefile:1575: _module_/tmp/tmp.quafAlVvP1] Error 2
make[1]: Leaving directory '/usr/src/kernels/4.17.2-200.fc28.x86_64'
make: *** [Makefile:147: build] Error 2
        Unable to build compatible LiKI module - you may still be able to use ftrace mode
```

**Comment by mhudec** [comment]
You might want to wait a bit until LinuxKI supports this new kernel (according to FAQ, it seems that kernels up to 4.16.3 are supported, you seem to have 4.17.2, that is quite recent...).

**Comment by cvoltz** [comment]
4.17.2 is the latest kernel for Fedora 28. I added the issue to let the developer know that changes will be needed to support the newer kernel.

**Comment by MarkCRay** [comment]
In 4.17 kernels, the getname() functions changed from 4 arguments to 3, as seen in the struct proto_ops.  
4.16.18:

	int		(*getname)   (struct socket *sock,
				      struct sockaddr *addr,
				      int *sockaddr_len, int peer);

4.17:

	int		(*getname)   (struct socket *sock,
				      struct sockaddr *addr,
				      int peer);

You can see the sockaddr_len parameter is no longer passed.

I'll need to update the LiKI code with some ifdefs and test.     As usual, there is always a lag between the latest Linux releases and when I can get around to making the changes, testing, and  releasing a new package.   But you do have access to the source in the /opt/linuxki/src directory.   Just go in and remove the 3rd argument to getname().   To generate a new LiKI DLKM module, remove the existing likit.ko files in /opt/linuxki/modules and execute /opt/linuxki/module_prep.   This should build with the latest source in /opt/linuxki/src.

Its hard to keep up with the ever-changing Linux kernels...

**Comment by MarkCRay** [comment]
This is fixed in LinuxKI version 5.5, released 08/13/2018.
